test/e2e/operator/nmstate_install_test.go

```diff
@@ -0,0 +1,135 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	v1 "k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nmstatev1beta1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1beta1"
+)
+
+func installNMState(nmstate nmstatev1beta1.NMState) {
+	_, err := getNMStateOperator(nmstate)
+	Expect(apierrors.IsNotFound(err)).To(BeTrue(), "NMState CR does not exist yet")
+	err = framework.Global.Client.Create(context.TODO(), &nmstate, &framework.CleanupOptions{})
+	Expect(err).ToNot(HaveOccurred(), "NMState CR created without error")
+}
+
+func uninstallNMState(nmstate nmstatev1beta1.NMState) {
+	_, err := getNMStateOperator(nmstate)
+	Expect(apierrors.IsNotFound(err)).To(BeFalse(), "NMState CR exists")
```

Comment:
You don't need to check existence to call delete, just call delete and if error is NotFound it returns also checking that it's really delted and Eventually helps too so we block until etcd delete it.

Example here:
https://github.com/nmstate/kubernetes-nmstate/blob/da83e5924b6348e8c506e157f83ec169171206cb/test/e2e/handler/utils.go#L167-L177